### DISCUSSION TITLE
how can I implement registering user functionality?

### DISCUSSION THREAD

**Discussion by Knorway** [directus/directus#12135] [Archive]
I'm reading [documentation](https://docs.directus.io/reference/authentication/) for hours but I can't find the proper way to register a user, not admin users for admin panel. it seems documentation is about admin users. 

I want to implement authentication like I make some app(e.g: social network) and let users register(local or OAuth) and save them to users collection(not directus_users). and then i give them back jwt token which will be attached on every request. 

only thing looks possible for me at this point is making extension exposes endpoint such as `'/auth/register-local'`, `'/auth/register-google/` and one for validate token.I succeed to implement it(I found directus extension system is very scalable and easy to add tho 👍) but it doesn't apply to default endpoints like `'/items/posts'/` cos extensions pass me down the only `router` and i can't access to main `express app`.

I feel like making custom endpoint for this isn't the way to go. I'm treating directus as a whole backend and I could be misunderstanding some concept since I'm new to CMS.  any comments would be much helpful! and thank you for the amazing product!

**Comment by rijkvanzanten** [comment]
Seeing that you'd like to use JWT based auth, and allow SSO logins etc etc, I would recommend simply using directus_users for these users as well, rather than trying to rebuild the auth system on a custom `users` table. That way, you get everything out of the box. You could create a new role for these app-users, that doesn't have access to the Directus app (meaning that they're still users in the API, but can't access the admin app) 🙂

**Reply by Knorway** [reply]
thank you for you kind suggestion! I'll definitely try it!! 😁
